### PR TITLE
[DT-21] implement install manager for DT

### PR DESCRIPTION
**Pull request by G0retZ**
**What does this PR do?**

   Adds InstallManager implementation for DT service layer.

**Database changed?**

   Yes

**Where should the reviewer start?**

- Better by commits

**How should this be manually tested?**

  This is not connected to UI, so essentially just run the app and make sure no new errors in logs

**What are the relevant tickets?**

  Tickets related to this pull-request: [DT-21](https://aptoide.atlassian.net/browse/DT-21)

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Database changed? If yes - Migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] New Kotlin code has unit tests
- [ ] New flows in presenters unit tests
- [ ] Mappers/Validators with any kind of logic unit tests
- [ ] Functional tests pass

### CODE REVIEW

File: install-manager/src/main/java/cm/aptoide/pt/install_manager/App.kt [hunk 1]
```diff
@@ -17,7 +17,7 @@ import cm.aptoide.pt.install_manager.dto.InstallPackageInfo
 interface App<D> {
   val packageName: String
   val installedVersionName: String?
-  val installedVersionCode: Int?
```
Comment (jdandrade) [inline install-manager/src/main/java/cm/aptoide/pt/install_manager/App.kt:20]: What is the reason for this change? I believe that an apps version code according to google documentation is a positive integer, meaning we don't actually need a Long value here? Just curious.

**Review by jdandrade — APPROVED** [review]

File: install-manager/src/main/java/cm/aptoide/pt/install_manager/App.kt (hunk 1 above)
Comment (G0retZ) [reply install-manager/src/main/java/cm/aptoide/pt/install_manager/App.kt:20]: Google changed the API for this since SDK `28` now it is Long.
https://developer.android.com/reference/android/content/pm/PackageInfo#getLongVersionCode()

Comment (jdandrade) [reply install-manager/src/main/java/cm/aptoide/pt/install_manager/App.kt:20]: Good to know, makes sense, thanks.
